setup/classes/class.ilCtrlStructureReader.php

```diff
@@ -100,8 +101,11 @@ function readStructure($a_force = false, $a_dir = "", $a_comp_prefix = "",
 	*/
 	function read($a_cdir)
 	{
-		global $ilDB, $lng;
-		
+		global $lng;
```

Comment:
Please remove this global as you did with `$ilDB`. Are there any other?

Reply:
Where am I to pull $lng from? any hints?

Reply:
Background: It is a common fault of administrators to just unzip ILIAS packages into existing ILIAS code directories. Which leads to the existence of old code files "mostly" overwritten by new code files. But if not included in the new packages, old files will persist. I expect this not only being a problem for the structure reader but also for the artifact reader.

Anyway, I agree that administrators should be able to cope with an English error message, so removing the $lng dependency completely is fine for me.